src/main/java/com/bob/jr/TextToSpeech/MemberVoiceConfig.java

```diff
@@ -24,30 +24,78 @@ public class MemberVoiceConfig {
     private static final AudioConfig defaultAudioConfig = defaultAudioConfigBuilder.build();
 
+    transient private final VoiceSelectionParams.Builder voiceSelectionParamsBuilder;
+    transient private final AudioConfig.Builder audioConfigBuilder;
+    private VoiceSelectionParams voiceSelectionParams;
+    private AudioConfig audioConfig;
+
     public MemberVoiceConfig() {
-        this(defaultVoiceSelectionParams, defaultAudioConfig);
+        this(Optional.of(defaultVoiceSelectionParams), Optional.of(defaultAudioConfig));
     }
 
-    public MemberVoiceConfig(VoiceSelectionParams voiceSelectionParams, AudioConfig audioConfig) {
-        this.voiceSelectionParams = voiceSelectionParams;
-        this.audioConfig = audioConfig;
-        this.voiceSelectionParamsBuilder = defaultVoiceSelectionParamsBuilder.clone();
-        this.audioConfigBuilder = defaultAudioConfigBuilder.clone();
+    public MemberVoiceConfig(final Optional<VoiceSelectionParams> voiceSelectionParamsOptional, final Optional<AudioConfig> audioConfigOptional) {
+        final var voiceSelectionParams = voiceSelectionParamsOptional.orElse(defaultVoiceSelectionParams);
+        final var audioConfig = audioConfigOptional.orElse(defaultAudioConfig);
+
+        this.voiceSelectionParamsBuilder = defaultVoiceSelectionParamsBuilder.clone().mergeFrom(voiceSelectionParams);
+        this.audioConfigBuilder = defaultAudioConfigBuilder.clone().mergeFrom(audioConfig);
+        this.voiceSelectionParams = voiceSelectionParamsBuilder.build();
+        this.audioConfig = audioConfigBuilder.build();
+    }
+
+    public static MemberVoiceConfig unwrapMessageBuffer(final ByteArrayInputStream byteArrayInputStream) throws IOException {
+        final var unpacker = MessagePack.newDefaultUnpacker(byteArrayInputStream);
+        var paramCount = 0;
+
+        Optional<VoiceSelectionParams> voiceSelectionParamsOptional = Optional.empty();
+        Optional<AudioConfig> audioConfigOptional = Optional.empty();
+        var nilReached = false;
+        while (unpacker.hasNext() && !nilReached) {
+            final var format = unpacker.getNextFormat();
+            final var value = unpacker.unpackValue();
+
+            switch (format.getValueType()) {
+                case NIL:
+                    nilReached = true;
+                    break;
+                default:
+                    final var binaryValue = value.asBinaryValue().asByteArray();
+                    switch (paramCount) {
+                        case 0:
+                            voiceSelectionParamsOptional = Optional.of(VoiceSelectionParams.parseFrom(binaryValue));
+                            break;
+                        case 1:
+                            audioConfigOptional = Optional.of(AudioConfig.parseFrom(binaryValue));
+                            break;
+                        default:
+                            throw new IOException("Unexpected data format; abandoning unpack.");
```

Comment:
Can you catch this and log it out?

Reply:
it should be caught by the method that's calling it.